lib/openssl/x509.rb

```diff
@@ -60,6 +60,109 @@ def to_h # {"oid"=>sn|ln, "value"=>value, "critical"=>true|false}
       def to_a
         [ self.oid, self.value, self.critical? ]
       end
+
+      module Helpers
+        def find_extension(oid)
+          extensions.find { |e| e.oid == oid }
+        end
+
+        def x509_name_from_general_name(gn_asn1)
+          unless gn_asn1.tag_class == :CONTEXT_SPECIFIC
+            raise ASN1::ASN1Error "invalid GeneralName"
+          end
+
+          if gn_asn1.tag == 4
+            Name.new(gn_asn1.value.first.to_der)
+          else
+            # TODO: raise error instead of returning nil?
+            # this is some other kind of general name.
+            nil
```

Comment:
I think it's exception: It's not empty, but it's also exception to expected value. Even if our "expected value" is pretty limited at this time.